references/identity-facade/pipeline.sh

```diff
@@ -0,0 +1,356 @@
+#!/bin/sh
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+#################################
+### function: set_idp_env_var ###
+#################################
+set_idp_env_var() {
+
+    # discovery document of an OIDC compliant IdP
+    idp_discovery_document=https://"$APIGEE_ORG"-"$APIGEE_ENV".apigee.net/v1/openid-connect/.well-known/openid-configuration
+
+    # retrieve configuration data from a discovery document
+    response=$(curl --silent -k1 -fsSL -X GET -H "Accept:application/json" "$idp_discovery_document")
+    if [ "$( printf '%s' "$response" | grep -c error )" -ne 0  ]; then
+        echo "$response"
+        
+        exit 1
+    fi
+
+    # extract data used to feed the kvm
+    issuer=$( printf '%s' "$response" | jq .issuer )
+    authorization_endpoint=$( printf '%s' "$response" | jq .authorization_endpoint )
+    token_endpoint=$( printf '%s' "$response" | jq .token_endpoint )
+    jwks_uri=$( printf '%s' "$response" | jq .jwks_uri )
+    userinfo_endpoint=$( printf '%s' "$response" | jq .userinfo_endpoint )
+
+    # set env variables for kvm (idpConfig)
+    TEST_IDP_ISSUER=$(printf '%s' "$issuer" | awk -F\" '{print $2}' | awk -F\" '{print $1}')
+    export TEST_IDP_ISSUER
+    
+    TEST_IDP_APIGEE_REDIRECT_URI="https://$APIGEE_ORG-$APIGEE_ENV.apigee.net/v1/oauth20/callback"
+    export TEST_IDP_APIGEE_REDIRECT_URI
+    
+    TEST_IDP_AZ_HOSTNAME=$(printf '%s' "$authorization_endpoint" | awk -F\"https:// '{print $2}' | awk -F\" '{print $1}' | awk -F/ '{print $1}')
```

Comment:
here is a short way
`echo "$url" | sed "s/https:\/\/\(.*\)[\/+]/\1/"`

but you can use whatever you think it best

Reply:
https://stackoverflow.com/questions/2497215/how-to-extract-domain-name-from-url

have a look at what is easiest for beginners to understand - maybe something from the second answer